state/protocol/badger/mutator.go

```diff
@@ -480,8 +483,14 @@ func (m *FollowerState) Finalize(ctx context.Context, blockID flow.Identifier) e
 	}
 	header := block.Header
 
-	// FIRST: verify that the parent block is the latest finalized block. This
-	// must be the case, as the `Finalize(..)` method only finalizes one block
+	// keep track of metrics updates and protocol events to emit:
+	// * metrics are updated after a successful database update
+	// * protocol events are emitted atomically with the database update
```

Comment:
include toDo to move events out of the DB transaction

Reply:
Added TODOs: https://github.com/onflow/flow-go/pull/2625/commits/a562d1f08f0d4c1811887edab1ba309c9230d994